sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3841,6 +3840,42 @@ pi_result piQueueGetInfo(pi_queue Queue, pi_queue_info ParamName,
   case PI_QUEUE_INFO_DEVICE_DEFAULT:
     die("PI_QUEUE_INFO_DEVICE_DEFAULT in piQueueGetInfo not implemented\n");
     break;
+  case PI_QUEUE_INFO_STATUS: {
+    // If there are open command lists then submit them for execution.
+    {
+      std::unique_lock<pi_shared_mutex> Lock(Queue->Mutex);
+      PI_CALL(Queue->executeAllOpenCommandLists());
+    }
```

Comment:
why do we need/want this side-affect?

Reply:
For example, let's say user submitted 4 kernels and they all end up in the open command list which is not submitted for execution yet. 

After we check status of the L0 queue using zeCommandQueueSynchronize after that, it will report that queue is completed because it is empty. So we need to flush to know whether  all tasks in the queue are actually completed.

Another option will be just to report that queue is not finished if there is any open command list and its event list is not empty - meaning that there are some commands in the queue which are not event submitted to the L0 queue yet. Do you think this option is better?

Reply:
Ok, thanks, fixed.